File: individus-backend/src/main/java/cocktail/ng/individus/Individu.java
package cocktail.ng.individus;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
public class Individu {

    @Id
    private String id;
    private String nom;
    private String prenom;

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public static class Adresse {

        private String cp;

        public String getCp() {
            return cp;
        }

        public void setCp(String cp) {
            this.cp = cp;
        }
    }


}

File: demo/src/app/individus/individu.ts


import {Type} from 'class-transformer';
export class Individu  {
  id: string;
  nom: string;
  prenom: string;

  @Type(() => Adresse)
  adresse: Adresse;

  get nomEtPrenom() {
    return `${this.nom} ${this.prenom}`;
  }
}


export class Adresse {
  cp: string;
}
